src/mol-io/common/string-like.ts

```diff
@@ -0,0 +1,291 @@
+/**
+ * Copyright (c) 2025 mol* contributors, licensed under MIT, See LICENSE file for more info.
+ *
+ * @author Adam Midlik <midlik@gmail.com>
+ */
+
+import { utf8Read } from './utf8';
+
+
+/** Generalized string type */
+export type StringLike = string | String | CustomString
+
+export const StringLike = {
+    /** Return true if `obj` is instance of `StringLike` */
+    is(obj: unknown): obj is StringLike {
+        return typeof obj === 'string' || obj instanceof String || (obj as CustomString)._StringLike_ || false;
+    },
+
+    /** Try to convert `StringLike` to a primitive `string`. Might fail if the content is longer that max allowed string length. */
+    toString(str: StringLike): string {
+        try {
+            return str.toString();
+        } catch (err) {
+            throw new Error(`Failed to convert StringLike object into string. This might be because the length ${str.length} exceeds maximum allowed string length ${MAX_STRING_LENGTH}. (${err})`);
+        }
+    },
+};
+
+
+/** Essential subset of `string` functionality. Add more string methods if needed. */
+interface CustomString {
+    /** Flag for recognizing `StringLike` objects */
+    _StringLike_: true,
```

Comment:
`CustomString` is a strict subset of `string` (if we omit the `_StringLike_` magic flag which I don't think we need. So how about just doing this?

```suggestion
export const StringLike = {
    /** Try to convert `StringLike` to a primitive `string`. Might fail if the content is longer that max allowed string length. */
    asString(str: any): string {
        try {
            if (typeof str === 'string') return str;
            return str.toString();
        } catch (err) {
            throw new Error(`Failed to convert StringLike object into string. This might be because the length ${str.length} exceeds maximum allowed string length ${MAX_STRING_LENGTH}. (${err})`);
        }
    },
};


/** Essential subset of `string` functionality. Add more string methods if needed. */
export interface StringLike {
```

Reply:
In many places in the code we need to check whether something is `StringLike` (mainly in functions that take `StringLike | Uint8Array`).
Originally there was `if (typeof obj === 'string')...` , which i had to change to `if (StringLike.is(obj))...`.

How would you achieve this without the `_StringLike_` magic flag?

Reply:
How about check if it has a `charCodeAt` function and implement the .is function based on that?

Reply:
Alternatively, the actual implementations of the interface could have the magic flag, but not make it part of the interface to make it trivially compatible with string.

The .is check could then check if the value is a builtin string object or has the magic flag.

Reply:
> Alternatively, the actual implementations of the interface could have the magic flag, but not make it part of the interface to make it trivially compatible with string.
> 
> The .is check could then check if the value is a builtin string object or has the magic flag.

That does not force implementations to have the flag.

I think i can do the ducktyping approach (`typeof (obj as any).charCodeAt === 'function'`)

Reply:
Changed